frontend/src/machines/billings/billingMachine.ts

```diff
@@ -0,0 +1,553 @@
+        billingDetailError: undefined;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingDetailMember[];
+        };
+        submitBillingError: undefined;
+      };
+    }
+  | {
+      value: "submitBillingError";
+      context: Context & {
+        billings: Billing[];
+        billingError: undefined;
+        formMode: undefined;
+        members: GetMemberListResponse["data"];
+        membersError: undefined;
+        billingDetail: BillingDetail;
+        billingDetailError: undefined;
+        billingForm: BillingForm & {
+          title: string;
+          total: number;
+          chargedMember: string;
+          isBillEqual: boolean;
+          members: BillingDetailMember[];
+        };
+        submitBillingError: string;
+      };
+    };
+
+type MachineEvents =
+  | { type: "FETCH_BILLING" }
+  | { type: "FETCH_BILLING_SUCCES"; billingsData: Billing[] }
+  | { type: "FETCH_BILLING_ERROR"; billingsErrorMessage: string }
+  | { type: "ACTIVATE_BILLING_FORM" }
+  | { type: "REFETCH_BILLING" }
+  | { type: "FETCH_MEMBER_SUCCESS"; membersData: GetMemberListResponse["data"] }
+  | { type: "FETCH_MEMBER_ERROR"; membersErrorMessage: string }
+  | { type: "REFTECH_MEMBERS" }
+  | { type: "FETCH_BILLING_DETAIL_SUCCES"; billingDetailData: BillingDetail }
+  | { type: "FETCH_BILLING_DETAIL_ERROR"; billingDetailErrorMessage: string }
+  | { type: "REFTECH_BILLING_DETAIL" }
+  | {
+      type: "NEXT_STEP";
+      billingName: string;
+      total: number;
+      chargedMember: string;
+    }
+  | { type: "PREV_STEP" }
+  | {
+      type: "SUBMIT_BILLING";
+      isBillEqual: boolean;
+      members: GetMemberListResponse["data"];
+    }
+  | { type: "SUBMIT_BILLING_SUCCES" }
+  | { type: "SUBMIT_BILLING_ERROR"; submitBillingErrorMessage: string }
+  | { type: "BACK_TO_BILLING_SCREEN" }
+  | { type: "BACK_TO_SECOND_STEP_FORM" }
+  | { type: "REFETCH_SUBMIT_BILLING" };
+
+export const billingMachine = createMachine<
+  Context,
+  MachineEvents,
+  MachineState
+>({
+  schema: {
+    context: {} as Context,
+    events: {} as MachineEvents,
+  },
+  id: "billing",
+  initial: "idle",
+  context: {
+    billings: undefined,
+    billingError: "",
+    formMode: undefined,
+    members: undefined,
+    membersError: undefined,
+    billingDetail: undefined,
+    billingDetailError: "",
+    billingForm: undefined,
+    submitBillingError: undefined,
+  },
+  states: {
+    idle: {
+      on: {
+        FETCH_BILLING: "loadingBilling",
+      },
+    },
+    loadingBilling: {
+      on: {
+        FETCH_BILLING_SUCCES: {
+          target: "getBillingOK",
+          actions: "assignBillingsData",
+        },
+        FETCH_BILLING_ERROR: {
+          target: "getBillingError",
+          actions: "assignBillingsErrorMessage",
+        },
+      },
+    },
+    getBillingOK: {
+      entry: "assignBillingsData",
```

Comment:
entry ini buat apa ya dan ?

Reply:
Aku taruh entry di state `getBillingOK` biar setiap kali state tersebut dikunjungi, data billing nya bakal fresh terus.

In this case cocok buat finite state machine kita yang gada final states nya (semua bakal balik lagi ke state `getBillingOK`

Reply:
oh, jadi entry itu buat jalanin suatu side effect ketika masuk ke sebuah state ya ? menarik